src/logic/safe/utils/allSettled.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Helper function to handle multiple async calls
+ * It will return the expected value for each parameter passed when fulfilled or a `null` in case of an error
+ * @param args
+ * @returns ExpectedValues
+ */
+export const allSettled = async <ExpectedValues extends unknown[]>(...args: unknown[]): Promise<ExpectedValues> => {
+  const data = new Array(args.length).fill(null)
+  const promises = await Promise.allSettled(args)
+
+  for (let i = 0; i < promises.length; i++) {
```

Comment:
I'm more for the `promises.forEach` syntax ;)

Reply:
changed